src/Entity/NphOrder.php

```diff
@@ -268,4 +268,14 @@ public function canModify($type): bool
         }
         return false;
     }
+
+    public function getSampleGroupBySampleCode($sampleCode): string
+    {
+        foreach ($this->nphSamples as $nphSample) {
+            if ($nphSample->getSampleCode() === $sampleCode) {
+                return $nphSample->getSampleGroup();
+            }
+        }
+        return '';
```

Comment:
It seems odd to return an empty string.  Would this work if the function returned null?